### ISSUE TITLE
Verbatim theorem environment

### ISSUE THREAD

**Issue by amtssk** [rstudio/bookdown#923]
How to show verbatim the syntax of the theorem environment like this is done in "Bookdown: Authoring Books and Technical Documents with R Markdown"?   

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.org/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('bookdown')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/bookdown')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by tchevri** [comment]
I am not quite sure what you are asking, I think you want to show what you see in [https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html](https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html), section 2.2.2
If that's the case, then to show verbatim, simply fence of your theorem block with single ticks:  

````
` ```{theorem}  
Here is my theorem.
```
`
````

If you want to see the underlying latex or html code for pdf or html output as transformed by pandoc, simply fence the custom block with one more tick: 3+1 = 4:

`````
````
```{theorem}  
Here is my theorem.
```
````
`````

Then just compile to your desired format.
Apologies if I am wrong about what you are asking - it's always best if you provide MWE (in any case, note that to get the above, I added one extra tick, each time, so 4 for the first block and... 5 for the second block).  Many thanks, thomas

**Comment by tchevri** [comment]
btw - i think you are supposed to ask on SO first - as it says in your promise when filing an issue.... ` If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there. `

**Comment by amtssk** [comment]
Yes, I want exactly the same output as in the manual.  Actually, I had tried both your suggestions before asking for help. Your first solution puts everything in one line: 

```{theorem} Here is my theorem. ``` 

which is not quite the same.  When I try your second solution, the html output is this line 

\BeginKnitrBlock{theorem}<div class="theorem"><span class="theorem" id="thm:unnamed-chunk-1"><strong>(\#thm:unnamed-chunk-1) </strong></span>Here is my theorem.</div>\EndKnitrBlock{theorem}

which is also not quite what I want.  What has gone wrong? Thanks.

**Comment by amtssk** [comment]
For some reason the output for the second solution did not show up right. Here is my second attempt:

\BeginKnitrBlock{theorem}<div class="theorem"><span class="theorem" id="thm:unnamed-chunk-1"><strong>(\#thm:unnamed-chunk-1) </strong></span>Here is my theorem.</div>\EndKnitrBlock{theorem}

**Comment by tchevri** [comment]
for new lines - you need to have a double space at the end of the sentence.
So make sure you have the "  " after the theorem first line - when you copied, it must have been truncated.
This is very basic Markdown btw.
The approach with single ticks works just fine and delivers the result you seem to be seeking - i tried it on my end before responding and got the result.

I could be wrong - but this is not the appropriate forum for this type of question - as per your pledge - you promised to ask on stack overflow first - read your pledge, please.

many thanks, thomas

**Comment by amtssk** [comment]
Hi Thomas,

I am new to this I admit and I need to explain to others new to this (used
to Latex instead) how to use bookdown learning as little as possible as
they are very busy with other things now.

The double space does not help though.
````
```{theorem}
Here is my theorem.
```
````
still generates

\BeginKnitrBlock{theorem}<div class="theorem"><span class="theorem"
id="thm:unnamed-chunk-1"><strong>(\#thm:unnamed-chunk-1)
</strong></span>Here is my theorem.  </div>\EndKnitrBlock{theorem}




Cheers,
Serguei

**Comment by shenfei** [comment]
@amtssk usually we increase the number of backticks to get a high level code block, i.e. with "````" around the "```" block. But for theorem environment, there is a trick. The correct form is
`````
````markdown
`r ''````{theorem}
Here is my theorem.
```
````
`````

I’ve encountered the same problem before, but the good news is bookdown’s document is written in bookdown itself.
So we can learn the trick from the author’s [own examples](https://github.com/rstudio/bookdown/blob/master/inst/examples/02-components.Rmd#L232).

**Comment by amtssk** [comment]
Dear Shen,

Many thanks!!!
At last, I have got some understanding .
This does work for me in all three output formats. Though in EPUB the light
grey background has gone. Probably, I have to edit output.yml and fill in
the right options. Where do I find their description?

Through trial and error I have found that if I put "empty space" just in
front of theorem in {theorem}, it works too, but in PDF and EPUB, the
colour is B&W on white background.

Best wishes,
Serguei
